dissect/target/plugins/os/unix/log/auth.py

```diff
@@ -1,60 +1,362 @@
+from __future__ import annotations
+
+import itertools
+import logging
 import re
+from abc import ABC, abstractmethod
+from datetime import datetime
+from functools import lru_cache
 from itertools import chain
-from typing import Iterator
+from pathlib import Path
+from typing import Any, Iterator
 
+from dissect.target import Target
 from dissect.target.exceptions import UnsupportedPluginError
-from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.helpers.fsutil import TargetPath, open_decompress
+from dissect.target.helpers.record import DynamicDescriptor, TargetRecordDescriptor
 from dissect.target.helpers.utils import year_rollover_helper
-from dissect.target.plugin import Plugin, export
-
-AuthLogRecord = TargetRecordDescriptor(
-    "linux/log/auth",
-    [
-        ("datetime", "ts"),
-        ("string", "message"),
-        ("path", "source"),
-    ],
+from dissect.target.plugin import Plugin, alias, export
+
+log = logging.getLogger(__name__)
+
+_RE_TS = r"^[A-Za-z]{3}\s*[0-9]{1,2}\s[0-9]{1,2}:[0-9]{2}:[0-9]{2}"
+_RE_TS_ISO = r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{6}\+\d{2}:\d{2}"
+
+RE_TS = re.compile(_RE_TS)
+RE_TS_ISO = re.compile(_RE_TS_ISO)
+RE_LINE = re.compile(
+    rf"(?P<ts>{_RE_TS}|{_RE_TS_ISO})\s(?P<hostname>\S+)\s(?P<service>\S+?)(\[(?P<pid>\d+)\])?:\s(?P<message>.+)$"
```

Comment:
The two suggestions corresponding to the suggestions further in the code:

Making a non-capturing group (and stripping spaces):
```suggestion
    rf"(?:{_RE_TS}|{_RE_TS_ISO})\s(?P<hostname>\S+)\s(?P<service>\S+?)(\[(?P<pid>\d+)\])?:\s+(?P<message>.+?)\s*$"
```
Or just check for the similar ending part of the timestamps and use `RE_LINE.search(line)` in the code below
```suggestion
    rf"\d{2}:\d{2}\s(?P<hostname>\S+)\s(?P<service>\S+?)(\[(?P<pid>\d+)\])?:\s+(?P<message>.+?)\s*$"
```

Reply:
This regex is getting long, can you use `re.VERBOSE` to split it up?
https://docs.python.org/3/library/re.html#re.VERBOSE

Reply:
Used `re.VERBOSE` everywhere now